docs/how-to-work-on-coding-challenges.md

```diff
@@ -95,6 +97,209 @@ Optional Test tear down code.
+**Note:** If you are going to use an example code in YAML, use `yaml` instead of `yml` for the language to the right of the backticks.
+See example below:
+````
+The following is an example of code:
+
+```{language}
+
+[YOUR CODE HERE]
+
+```
+
+````
+- Additional information in the form of a note should be formatted `<strong>Note:</strong> Rest of note text...`
+- If multiple notes are needed, then list all of the notes in separate sentences using the format `<strong>Notes:</strong> First note text.  Second note text.`.
+- Use double quotes where applicable
+
+## Formatting seed code
+
+Here are specific formatting guidelines for the challenge seed code:
+
+- Use two spaces to indent
+- JavaScript statements end with a semicolon
+- Use double quotes where applicable
+- Comments made should have a space between the comment characters and the comment themselves
+
+    `// Fix this line`
+
+## Formatting Solution
+
+The challenge solution must be in proper markdown tags. The code must be wrapped in `section` tags with an id of `solution`.
+
+````
+<section id='solution'>
+
+```html or ```js
+ [ SOLUTION CODE]
+```
+
+</section>
+````
+
+Example:
+
+````
+<section id='solution'>
+
+```html
+<style>
+  #container p {
+    font-family: Arial, sans-serif;
+  }
+</style>
+
+<div id="container">
+  <p>Hello World!</p>
+</div>
+```
+
+</section>
+````
+
+## Why do we have all these rules?
+
+Our goal is to develop a fun and clear interactive learning experience.
+
+Designing interactive coding challenges is difficult. It would be much easier to write a lengthy explanation or to create a video tutorial, and there's a place for those on Medium and YouTube. However, for our core curriculum, we're sticking with what works best for most people - a fully interactive, video game-like experience.
+
+We want campers to achieve a flow state. We want them to build momentum and blast through our curriculum with as few snags as possible. We want them to go into the projects with confidence and a wide exposure to programming concepts.
+
+Creating these challenges requires immense creativity and attention to detail. But you'll have plenty of help. You have support from a whole team of contributors, whom you can bounce ideas off of and demo your challenges to. Stay active in the [contributors room](https://gitter.im/freecodecamp/contributors) and ask lots of questions.
+
+With your help, we can design an interactive coding curriculum that will help millions of people learn to code for years to come.
+
+### Curriculum Challenge Hints and Solutions
+
+With your help, we can create a comprehensive reference tool that will help millions of people who are learning to code, for years to come. Curriculum hint topics are located on [our forum](https://www.freecodecamp.org/forum/c/guide) under the `Guide` category.
```

Comment:
We will need to update this once we finalize a new category name for the hints/solutions topics.